core/src/main/scala/quasar/destination/gbq/GBQDestination.scala

```diff
@@ -0,0 +1,213 @@
+        .withHeaders(bearerToken)
+        .withContentType(`Content-Type`(MediaType.application.json))
+        .withEntity(dCfg)
+
+      client.fetch(datasetReq) { resp =>
+        resp.status match {
+          case Status.Ok | Status.Conflict =>
+            ().asRight[InitializationError[Json]].pure[F]
+          case status =>
+            DestinationError.malformedConfiguration(
+              (destinationType, jString("Reason: " + status.reason), 
+              sanitizedConfig.toString)).asLeft.pure[F]
+        }
+      }
+    }
+
+  private def mkGbqJob(
+      client: Client[F],
+      accessToken: AccessToken,
+      jCfg: GBQJobConfig): F[Either[InitializationError[Json], Uri]] = {
+
+    implicit def jobConfigEntityEncoder: EntityEncoder[F, GBQJobConfig] = jsonEncoderOf[F, GBQJobConfig]
+
+    val authToken = Authorization(Credentials.Token(AuthScheme.Bearer, accessToken.getTokenValue))
+    val jobReq = Request[F](
+      method = Method.POST,
+      uri = Uri.fromString(s"https://bigquery.googleapis.com/upload/bigquery/v2/projects/${jCfg.destinationTable.project}/jobs?uploadType=resumable").getOrElse(Uri()))
+        .withHeaders(authToken)
+        .withContentType(`Content-Type`(MediaType.application.json))
+        .withEntity(jCfg)
+
+    client.fetch(jobReq) { resp =>
+      resp.status match {
+        case Status.Ok => resp.headers match {
+          case Location(loc) =>
+            loc.uri.asRight.pure[F]
+        }
+        case _ =>
+          DestinationError.malformedConfiguration(
+            (destinationType, jString("Reason: " + resp.status.reason), 
+            sanitizedConfig.toString)).asLeft.pure[F]
+      }
+    }
+  }
+
+  private def upload(client: Client[F], bytes: Stream[F, Byte], uploadLocation: Uri, accessToken: AccessToken): Stream[F, Unit] = {
+    val bearerToken = Authorization(Credentials.Token(AuthScheme.Bearer, accessToken.getTokenValue))
+    val destReq = Request[F](method = Method.POST, uri = uploadLocation)
+        .withHeaders(bearerToken)
+        .withContentType(`Content-Type`(MediaType.application.`octet-stream`))
+        .withEntity(bytes)
+
+    val doUpload = client.fetch(destReq) { resp =>
+      resp.status match {
+        case Status.Ok => ().pure[F]
+        case _ => ApplicativeError[F, Throwable].raiseError[Unit](
+            new RuntimeException(s"Upload failed: ${resp.status.reason}" ))
+      }
+    }
+    Stream.eval(doUpload)
+  }
+}
+
+object GBQDestination {
+    def apply[F[_]: Concurrent: ContextShift: MonadResourceErr, C](client: Client[F], config: GBQConfig, sanitizedConfig: Json)
+        : Resource[F, Either[InitializationError[C], Destination[F]]] = {
+      val x: Either[InitializationError[C], Destination[F]] = new GBQDestination[F](client, config, sanitizedConfig).asRight[InitializationError[C]]
+      Resource.liftF(x.pure[F])
```

Comment:
Why is this a `Resource`? Is there anything in `GBQDestination` which needs to be cleaned up which *isn't* being cleaned up?